File: ViewPager/app/src/main/java/github/nisrulz/viewpager/Page.java
package github.nisrulz.viewpager;

import android.os.Bundle;
import androidx.fragment.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;


public class Page extends Fragment {

    // Store instance variables
    private String title;
    private int page;


    public Page() {
        // Required empty public constructor
    }


    public static Page newInstance(int page_number, String title) {
        Page fragment = new Page();
        Bundle args = new Bundle();
        args.putInt("page", page_number);
        args.putString("title", title);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        page = getArguments().getInt("page", 0);
        title = getArguments().getString("title");

    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_page, container, false);
        ((TextView) view.findViewById(R.id.text)).
                setText(page + " - " + title);
        return view;
    }
}

File: ViewPager/app/src/main/java/github/nisrulz/example/viewpager/Page.kt
package github.nisrulz.example.viewpager

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import github.nisrulz.example.viewpager.databinding.FragmentPageBinding

class Page : Fragment() {
    // Store instance variables
    private var title: String? = null
    private var page = 0

    private lateinit var binding: FragmentPageBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        arguments?.apply {
            page = getInt("page", 0)
            title = getString("title")
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentPageBinding.inflate(inflater, container, false)
        with(binding) {
            textView.text = "$page - $title"
        }

        return binding.root
    }

    companion object {
        fun newInstance(page_number: Int, title: String?): Page {
            val fragment = Page()
            val args = Bundle().apply {
                putInt("page", page_number)
                putString("title", title)
            }
            fragment.arguments = args
            return fragment
        }
    }
}